From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Wed, 15 Nov 2017 23:05:08 +0100
Subject: [PATCH] {BUG in 4.15-rc1} fs/xfs/scrub/quota: error may be
 uninitialized

clang reports:

    fs/xfs/scrub/quota.c:228:6: error: variable 'error' is used
    uninitialized whenever 'if' condition is true
          [-Werror,-Wsometimes-uninitialized]
            if (ip->i_d.di_flags & XFS_DIFLAG_REALTIME) {
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    fs/xfs/scrub/quota.c:299:9: note: uninitialized use occurs here
            return error;
                   ^~~~~
    fs/xfs/scrub/quota.c:228:2: note: remove the 'if' if its condition
    is always false
            if (ip->i_d.di_flags & XFS_DIFLAG_REALTIME) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    fs/xfs/scrub/quota.c:210:14: note: initialize the variable 'error'
    to silence this warning
            int                             error;
                                                 ^
                                                  = 0
---
 fs/xfs/scrub/quota.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/xfs/scrub/quota.c b/fs/xfs/scrub/quota.c
index 3d9037eceaf1..7be1bcbaa64d 100644
--- a/fs/xfs/scrub/quota.c
+++ b/fs/xfs/scrub/quota.c
@@ -227,6 +227,7 @@ xfs_scrub_quota(
 	xfs_ilock(ip, XFS_ILOCK_EXCL);
 	if (ip->i_d.di_flags & XFS_DIFLAG_REALTIME) {
 		xfs_scrub_ino_set_corrupt(sc, sc->ip->i_ino, NULL);
+		error = -ENOENT;
 		goto out_unlock_inode;
 	}
 	max_dqid_off = ((xfs_dqid_t)-1) / qi->qi_dqperchunk;
-- 
